From: Simon McVittie <smcv@collabora.com>
Date: Wed, 25 Mar 2020 13:07:08 +0000
Subject: common: Only run variant-schema-compiler once

A make rule like

    a b: x y
            command

does not mean "run command to generate a and b from x and y". Instead,
it means "run command to generate a from x and y", and, separately,
"run command to generate b from x and y". In a parallel build this
could mean that we try to run the variant-schema-compiler twice, in
parallel, with the output from each run overwriting the other.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Forwarded: https://github.com/flatpak/flatpak/pull/3489
---
 common/Makefile.am.inc | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/common/Makefile.am.inc b/common/Makefile.am.inc
index f7d93ad..4b74c03 100644
--- a/common/Makefile.am.inc
+++ b/common/Makefile.am.inc
@@ -75,9 +75,13 @@ nodist_libflatpak_common_base_la_SOURCES = \
 BUILT_SOURCES += $(nodist_libflatpak_common_base_la_SOURCES)
 CLEANFILES += $(nodist_libflatpak_common_base_la_SOURCES)
 
-common/flatpak-variant-private.h common/flatpak-variant-impl-private.h: variant-schema-compiler/variant-schema-compiler  data/flatpak-variants.gv
+common/flatpak-variant-private.h: variant-schema-compiler/variant-schema-compiler  data/flatpak-variants.gv
 	$(AM_V_GEN) $(srcdir)/variant-schema-compiler/variant-schema-compiler --outfile-header common/flatpak-variant-private.h  --outfile common/flatpak-variant-impl-private.h --prefix var $(srcdir)/data/flatpak-variants.gv
 
+# Generated by the same tool at the same time
+common/flatpak-variant-impl-private.h: common/flatpak-variant-private.h
+	@:
+
 libflatpak_common_base_la_SOURCES = \
 	common/flatpak-utils-base.c \
 	common/flatpak-utils-base-private.h \
